src/pages/workspaces/workspace/Data.js

```diff
@@ -569,17 +569,28 @@ const WorkspaceData = _.flow(
 
   const loadWdsSchema = async () => {
     if (isAzureWorkspace) {
-      // TODO: AJ-783: Logic will need to exist here to check if Leo app is past "Provisioning"
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const url = await wdsDataTableProvider.proxyUrlPromise
-        setProxyUrlLoaded(!!url)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
-      }
+      // get the proxy url for this app from Leo
+      setWdsProxyUrl({ status: 'Loading', state: '' })
+      await wdsDataTableProvider.proxyUrlPromise
+        .then(url => {
+          if (!!url) {
+            setWdsProxyUrl({ status: 'Ready', state: url })
```

Comment:
For some history, we initially had a proxyUrl/setProxyUrl in the data page, and shifted to have things inside the WDS provider to encapsulate WDS specific logic there. I like this change as far as making our WDS interactions more robust, and thinking about if there's a way to encapsulate these WDS interactions in the WDS provider.

Reply:
hm, yeah - I could see moving all this logic into `WdsDataTableProvider` and giving it a `refresh()` or `initialize()` method, and also giving it `isReady` / `isLoading` / `isError` member variables. The tricky part might be that `WdsDataTableProvider` doesn't have any React state to work with.